.github/workflows/update-catalogs.yml

```diff
@@ -0,0 +1,21 @@
+name: Update catalogs.json from PostgreSQL .dat files
+
+on:
+  push:
+    branches:
+      - main
+      - 'hotfix/**'
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  update_catalogs:
+    name: Update catalogs.json from PostgreSQL .dat files if necessary and create a pull request in that case
```

Comment:
Revisiting this, I still think it's more than fine to leave this update process to be manual, and to run it once a release (or if a user specifically complains):

* Once again, doing this without the accompanying type handler wouldn't be of much use to anyone, since the new type wouldn't actually be usable.
* In addition, your NoTypeLoadingVsTypeLoading test would catch any new type added to PG the moment we run tests against the new version beta. That would be a good signal to run the Perl script etc. (the test could even tell you to do it in the failure message)

Reply:
Yeah, I don't really care that much.

I mostly did it because you made a poll for it (https://github.com/npgsql/npgsql/pull/4419#discussion_r851800824) and I liked @vonzshik's [idea](https://github.com/npgsql/npgsql/pull/4419#discussion_r852018166).
Plus it is not very hard to do, once you have figured out the APIs for it.
We can also just run it once per month or whatever.

> doing this without the accompanying type handler wouldn't be of much use to anyone, since the new type wouldn't actually be usable

Well, if you look at the current json, you'll find out that this is already true for quite a few types we will have in `PostgresMinimalDatabaseInfo`, but mind you, we're not the only ones who could be implementing type handlers.